pkg/rehearse/rehearse.go

```diff
@@ -581,11 +586,7 @@ func setupDependencies(
 // Allow manipulating the speed of time for tests
 var second = time.Second
 
-func ensureImageStreamTags(ctx context.Context, client ctrlruntimeclient.Client, ists apihelper.ImageStreamTagMap, clusterName, namespace string, istImportClient ctrlruntimeclient.Client, log *logrus.Entry) error {
-	if clusterName == appCIContextName {
```

Comment:
This is the tricky part.
Let us verify this: Found out where the request is generated with `cluster` encoded. IIRC, it is `ci-operator`.
I believe we just need to do the mirror in one cluster because the target would be the same on QCI for the requests on different clusters.

Reply:
I don't understand why only from one cluster. I think as it is right now the `ImageStreamImport` created in the `test_images_distributor.go` imports the IS only to one cluster, not all.

Reply:
You can compare those on the cluster: The difference is `spec.clusterName`.

```console
oc --context build01 get testimagestreamtagimports  -n ci
oc --context app.ci get testimagestreamtagimports  -n ci
```

Reply:
Let us track down `testimagestreamtagimports` together.
Producer 1: the rehearsal tool

https://github.com/openshift/ci-tools/blob/7f1ce7102624689fe0a1d695232ea5e2266c7ef2/pkg/rehearse/rehearse.go#L608-L620

It sets up its `.spec.cluster` with a loop for each build-farm.

Producer 2: `ci-operator`

https://github.com/openshift/ci-tools/blob/7f1ce7102624689fe0a1d695232ea5e2266c7ef2/pkg/defaults/defaults.go#L1057-L1072

It does not set up its `.spec.cluster` and it creates only one object on the cluster where it is running.

Consumer: `test_images_distributor`. As displayed between, they have different handlers accordingly for the above two cases.

https://github.com/openshift/ci-tools/blob/7f1ce7102624689fe0a1d695232ea5e2266c7ef2/pkg/controller/test-images-distributor/test_images_distributor.go#L90-L104

One more thing (not relevant to this PR tho): There are two sources of the events to those handlers.

https://github.com/openshift/ci-tools/blob/7f1ce7102624689fe0a1d695232ea5e2266c7ef2/pkg/controller/test-images-distributor/test_images_distributor.go#L115-L132

I know this is quite complicated (i must admit I might have imcomplete/wrong understanding even though I have read the code many rounds, including the one for write this comment down).
So the general idea is:
When we need to create an import on a tag on a cluster, we create `testimagestreamtagimports` which is watched by `test_images_distributor` which makes the import takes place.
That is what the producer does, including the rehearsal tool.
Your task is ... make sure the tag is available on QCI by oc-mirror cmd.
You need to do it only once because the target image is the same image on QCI.